### PR TITLE
PP-12444 simplified settings - stripe details index

### PR DESCRIPTION
**Pull request by nlsteers**
## WHAT

- new settings page for stripe details
  - index view implementation
  - [temp] backport task list component css from DSv5 to DSv4
  - new utility function to format stripe setup tasks from connector progress object
- enforce live account only for some settings when service has gone live
- fix issue where users were shown an error page instead of being redirected to the login page when session had expired on simplified account routes
- DRY up system messages nunjucks macro
  - implement multi-line messages
- new simplified settings page layout
  - update existing simplified settings templates to use new layout

### SCREENS

Screenshot 2024-10-24 at 17 01 09

Screenshot 2024-10-25 at 12 42 56

### CODE REVIEW

File: app/views/includes/system-messages.njk [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{% macro systemMessages(params) %}
```
Comment (oswaldquek) [inline app/views/includes/system-messages.njk:1]: Be good to show a screenshot of this file in the PR.

File: app/views/simplified-account/settings/stripe-details/index.njk [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        </strong>
+      {% elseif value.status == true %}
+        <strong class="govuk-tag">
+          Completed
+        </strong>
+      {% else %}
+        <strong class="govuk-tag govuk-tag--blue">
+          Not yet started
+        </strong>
+      {% endif %}
```
Comment (oswaldquek) [inline app/views/simplified-account/settings/stripe-details/index.njk:32]: Could be for a next PR but should there be an "In progress" status?

**Review by oswaldquek — COMMENTED** [review]
Some questions.

File: app/controllers/simplified-account/settings/stripe-details/stripe-details.controller.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+const friendlyStripeTasks = require('../../../../utils/simplified-account/settings/stripe-details/tasks')
+
+async function get (req, res) {
+  const account = req.account
+  const service = req.service
+  const stripeDetailsTasks = friendlyStripeTasks(account, service)
+  return response(req, res, 'simplified-account/settings/stripe-details/index', {
+    messages: res.locals?.flash?.messages ?? [],
+    stripeDetailsTasks,
+    incompleteTasks: Object.values(stripeDetailsTasks).some(task => task.status === false)
```
[the remark this answers is not in the document]
Comment (nlsteers) [reply app/controllers/simplified-account/settings/stripe-details/stripe-details.controller.js:11]: yeah it's not in use yet but will be for dynamically deciding what text to show the user on the index view (if all tasks are complete we show a different paragraph), I will be adding this in another PR

File: app/utils/simplified-account/settings/stripe-details/tasks.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+const formatSimplifiedAccountPathsFor = require('../../format/format-simplified-account-paths-for')
+const paths = require('../../../../paths')
+
+const stripeDetailsTasks = Object.freeze({
```
[the remark this answers is not in the document]
Comment (nlsteers) [reply app/utils/simplified-account/settings/stripe-details/tasks.js:4]: ah okay, was trying to come up with something enum-like but it's not strictly an issue

File: app/views/includes/system-messages.njk (hunk 1 above)
Comment (nlsteers) [reply app/views/includes/system-messages.njk:1]: added

File: app/views/simplified-account/settings/stripe-details/index.njk (hunk 2 above)
Comment (nlsteers) [reply app/views/simplified-account/settings/stripe-details/index.njk:32]: yes there should but I haven't decided exactly how we'll track in progress tasks yet so the status is not implemented

**Review by oswaldquek — APPROVED** [review]

**Pull request merged**